fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MaterializedViewRewriter.java

```diff
@@ -1351,29 +1351,41 @@ private OptExpression tryUnionRewrite(RewriteContext rewriteContext,
         ScalarOperator mvOtherPreds = MvUtils.canonizePredicate(Utils.compoundAnd(
                 mvCompensationToQuery.getRangePredicates(),
                 mvCompensationToQuery.getResidualPredicates()));
+        List<ScalarOperator> rewrittenFailedPredicates = Lists.newArrayList();
```

Comment:
Can we move `rewrittenFailedPredicates `  to `EquationRewriter` as `AggregateFunctionRewriter`, so can simplify this retry policy?

Reply:
yes, but scan operator will add columnsToEnforce into getColRefToColumnMetaMap, not only use columnsToEnforce.

Reply:
here I just collect rewrittenFailedPredicates for union rewrite and try to rewrite it in tryRewritePredicates.